template_feed/Microsoft.DotNet.Common.ItemTemplates/content/Class-CSharp/.template.config/template.json

```diff
@@ -0,0 +1,75 @@
+{
+  "$schema": "http://json.schemastore.org/template",
+  "author": "Microsoft",
+  "classifications": [
+    "Common"
+  ],
+  "name": "Class",
+  "generatorVersions": "[1.0.0.0-*)",
+  "description": "A template for creating a class item",
+  "tags": {
+    "language": "C#",
+    "type": "item"
+  },
+  "groupIdentity": "DotNet.Common.Class",
+  "precedence": "0",
+  "identity": "DotNet.Common.Class.CSharp",
+  "shortName": "class",
+  "sourceName": "Class1",
+  "primaryOutputs": [
+    {
+      "path": "Class1.cs"
+    }
+  ],
+  "symbols": {
+    "DefaultNamespace": {
+      "type": "bind",
+      "binding": "msbuild:RootNamespace",
+      "replaces": "Company.ClassLibrary1"
+    },
+    "TargetFramework": {
+      "type": "bind",
+      "binding": "msbuild:TargetFramework"
+    },
+    "net6orLater": {
+      "type": "generated",
```

Comment:
One thing I notice here is that we have these complex conditions that are fairly easily expressed in MSBuild. In this case, `$([MSBuild]::IsTargetFrameworkCompatible($(TargetFramework), 'net6.0'))` would give what we want. Should the MSBuild binding mechanism be extended to allow evaluation of expressions?

Reply:
That's a nice idea, let's add the item to templating backlog on that.

Reply:
I have created an item
https://github.com/dotnet/templating/issues/5767